include/deal.II/base/vectorization.h

```diff
@@ -976,6 +1046,14 @@ class VectorizedArray<double, 8>
     data                = _mm512_i32gather_pd(index, base_ptr, 8);
   }
 
+  DEAL_II_ALWAYS_INLINE
+  void
+  gather(const std::array<double *, 8> &base_ptr, const unsigned int offset)
+  {
+    for (unsigned int v = 0; v < 8; v++)
```

Comment:
why do you need all these specializations of ``gather`` even though their implementation is identical?

Reply:
The problem is that not the function is specialized but the whole class...

Reply:
I guess the question should rather be: Do we really need gather to be a member function in case we cannot go to machine-specific code (and I guess there will not be this kind of intrinsics on the foreseeable architectures), or should we aim for providing a free-standing function `gather` (maybe within some namespace) that can use a single implementation?